a10_neutron_lbaas/v2/handler_lb.py

```diff
@@ -67,23 +67,27 @@ def delete(self, context, lb):
 
     def stats(self, context, lb):
         with a10.A10Context(self, context, lb) as c:
+            total_fwd_bytes, total_rev_bytes, curr_conns, tot_conns = 0, 0, 0, 0
             try:
                 name = self.meta(lb, 'id', lb.id)
                 r = c.client.slb.virtual_server.stats(name)
-
+                for port in r["virtual-server"]["port-list"]:
+                    total_fwd_bytes += port['stats']['total_fwd_bytes']
+                    total_rev_bytes += port['stats']['total_rev_bytes']
+                    curr_conns += port["stats"]["curr_conn"]
+                    tot_conns += port["stats"]["total_conn"]
                 return {
-                    "bytes_in": r["virtual_server_stat"]["req_bytes"],
-                    "bytes_out": r["virtual_server_stat"]["resp_bytes"],
-                    "active_connections":
-                        r["virtual_server_stat"]["cur_conns"],
-                    "total_connections": r["virtual_server_stat"]["tot_conns"]
+                    "bytes_in": total_fwd_bytes,
+                    "bytes_out": total_rev_bytes,
+                    "active_connections": curr_conns,
+                    "total_connections": tot_conns
                 }
             except Exception:
```

Comment:
Yes catching all exceptions can be problematic. I propose catching only NoneType errors only. The only other exceptions that are likely to be raised are AcosErrors and KeyErrors. This, and perhaps additional/enhanced testing, will give us greater visibility of further issues caused by future upstream changes.